FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Jobs/FhirToDataLakeProcessingJobSpliter.cs

```diff
@@ -0,0 +1,322 @@
+                new KeyValuePair<string, string>(FhirApiConstants.PageCountKey, FhirApiPageCount.Single.ToString("d")),
+                new KeyValuePair<string, string>(FhirApiConstants.TypeKey, resourceType),
+                isDescending
+                ? new KeyValuePair<string, string>(FhirApiConstants.SortKey, LastUpdatedApiParameterDesc)
+                : new KeyValuePair<string, string>(FhirApiConstants.SortKey, LastUpdatedApiParameter),
+            };
+
+            if (start != null)
+            {
+                parameters.Add(new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"ge{((DateTimeOffset)start).ToInstantString()}"));
+            }
+
+            var searchOptions = new BaseSearchOptions(null, parameters);
+
+            string fhirBundleResult = await _dataClient.SearchAsync(searchOptions, cancellationToken);
+
+            // Parse bundle result.
+            JObject fhirBundleObject;
+            try
+            {
+                fhirBundleObject = JObject.Parse(fhirBundleResult);
+            }
+            catch (JsonReaderException exception)
+            {
+                string reason = string.Format(
+                        "Failed to parse fhir search result for '{0}' with search parameters '{1}'.",
+                        searchOptions.ResourceType,
+                        string.Join(", ", searchOptions.QueryParameters.Select(parameter => $"{parameter.Key}: {parameter.Value}")));
+
+                _diagnosticLogger.LogError(reason);
+                _logger.LogInformation(exception, reason);
+                throw new FhirDataParseException(reason, exception);
+            }
+
+            List<JObject> fhirResources = FhirBundleParser.ExtractResourcesFromBundle(fhirBundleObject).ToList();
+
+            if (fhirResources.Any() && fhirResources.First().GetLastUpdated() != null)
+            {
+                return fhirResources.First().GetLastUpdated();
+            }
+
+            return null;
+        }
+
+        private async Task<DateTimeOffset?> GetNextSplitTimestamp(string resourceType, DateTimeOffset? start, SortedDictionary<DateTimeOffset, int> anchorList, CancellationToken cancellationToken)
+        {
+            int baseSize = start == null ? 0 : anchorList[(DateTimeOffset)start];
+            var last = start;
+            foreach (var item in anchorList)
+            {
+                var anchorValue = item.Value;
+                if (anchorValue == int.MaxValue)
+                {
+                    // Get resource count failed before, re-try to get resource count.
+                    var resourceCount = await GetResourceCountAsync(resourceType, last, item.Key, cancellationToken);
+                    var lastAnchorValue = last == null ? 0 : anchorList[(DateTimeOffset)last];
+                    anchorList[item.Key] = resourceCount == int.MaxValue ? int.MaxValue : resourceCount + lastAnchorValue;
```

Comment:
seems we can remove `anchorValue` and use `anchorList[item.Key]` directly?

Reply:
thanks, updated it.